client/settings/payment-methods-selector/index.js

```diff
@@ -0,0 +1,114 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, CheckboxControl, Icon, Modal } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { HorizontalRule } from '@wordpress/primitives';
+
+/**
+ * Internal dependencies
+ */
+import { addSelectedPaymentMethods } from 'data';
+import PaymentMethodIcon from 'settings/payment-method-icon';
+import './style.scss';
+
+const PaymentMethodsSelector = ( props ) => {
+	const { onClose, enabledPaymentMethods = [] } = props;
+	const [ paymentMethods, setPaymentMethods ] = useState(
+		enabledPaymentMethods.reduce(
+			( acc, value ) => {
+				acc[ value ] = true;
+				return acc;
+			},
+			{ giropay: false, sofort: false, sepa: false }
+		)
+	);
+
+	const onChangePaymentMethod = ( paymentMethod ) => {
+		return ( enabled ) => {
+			setPaymentMethods( {
+				...paymentMethods,
+				[ paymentMethod ]: enabled,
+			} );
+		};
+	};
+
+	const onAddSelected = () => {
```

Comment:
What do you think about naming the event handlers `handle{EventName}`? So in this case it would be called `handleAddSelected`. But I would rename `onChangePaymentMethod` to `makeHandlePaymentMethodChange`, since it's _curry-ing_

Reply:
Sounds good!